langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchOrganicResult.java

```diff
@@ -0,0 +1,204 @@
+package dev.langchain4j.web.search;
+
+import dev.langchain4j.data.document.Document;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.segment.TextSegment;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static dev.langchain4j.internal.Utils.getOrDefault;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+
+/**
+ * Represents an organic search results are the web pages that are returned by the search engine in response to a search query.
+ * This includes the title, link, content, and metadata of the web page.
+ * <p>
+ * These results are typically ranked by relevance to the search query.
+ * <p>
+ */
+public class WebSearchOrganicResult {
+
+    private final String title;
+    private final String link;
+    private final String content;
+    private final Map<String, String> resultMetadata;
+
+    /**
+     * Constructs a WebSearchOrganicResult object with the given content.
+     *
+     * @param content The content of the search result.
+     */
+    public WebSearchOrganicResult(String content) {
+        this(null, null, content, null);
+    }
+
+    /**
+     * Constructs a WebSearchOrganicResult object with the given content and link.
+     *
+     * @param content The content of the search result.
+     * @param link    The link associated with the search result.
+     */
+    public WebSearchOrganicResult(String content, String link) {
```

Comment:
Please order parameters consistently. In this ctor order is content, then link.
In other ctors it is link, then content.
I would propose to put parameters in the order of importance and optionality, so in this case:
link -> title -> content -> metadata ?